sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -456,36 +454,62 @@ def _x_else_x_o(self, x: Optional[Array]) -> Array:
         else:
             return self.x_o
 
+    def _process_x(self, x: Tensor) -> Tensor:
+        """Return x after casting to batched tensor and maybe comparing to x_o shape."""
```

Comment:
I think the casting should go in the caller (single line) and this function should be called e.g. `ensure_x_consistent_with_x_o`, because that's what I would like to read in the caller (the important thing). YOu can even do (in the caller):
```
x = self._ensure_consistency_with_x_o(atleast_2d_tensor(x))
```